sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -0,0 +1,235 @@
+
+            var requestOptions = new QueryAggregateSeriesRequestOptions();
+            requestOptions.InlineVariables["Temperature"] = numericVariable;
+            requestOptions.ProjectedVariables.Add("Temperature");
+
+            QueryAnalyzer queryAggregateSeriesAnalyzer = client.Query.CreateAggregateSeriesQueryAnalyzer(
+                tsId,
+                TimeSpan.FromSeconds(2),
+                TimeSpan.FromSeconds(30),
+                null,
+                requestOptions);
+
+            await foreach (TimeSeriesPoint point in queryAggregateSeriesAnalyzer.GetResultsAsync())
+            {
+                double? averageTemperature = (double?)point.GetValue("Temperature");
+                if (averageTemperature != null)
+                {
+                    Console.WriteLine($"{point.Timestamp} - Average temperature: {averageTemperature}.");
+                }
+            }
+            #endregion Snippet:TimeSeriesInsightsSampleQueryAggregateSeriesWithNumericVariable
+
+            #region Snippet:TimeSeriesInsightsSampleQueryAggregateSeriesWithAggregateVariable
+            Console.WriteLine("\n\nCount the number of temperature vents over the past 3 minutes, in 1-minute time slots.\n");
+
+            // Get the count of events in 60-second time slots over the past 3 minutes
+            DateTimeOffset now = DateTime.UtcNow;
+            DateTimeOffset endTime = now.AddMinutes(3);
+            DateTimeOffset startTime = now.AddMinutes(-3);
+
+            var aggregateVariable = new AggregateVariable(
+                new TimeSeriesExpression("count()"));
+
+            var aggregateSeriesRequestOptions = new QueryAggregateSeriesRequestOptions();
+            aggregateSeriesRequestOptions.InlineVariables["Count"] = aggregateVariable;
+            aggregateSeriesRequestOptions.ProjectedVariables.Add("Count");
+
+            QueryAnalyzer aggregateSeriesQueryAnalyzer = client.Query.CreateAggregateSeriesQueryAnalyzer(
+                tsId,
+                startTime,
+                endTime,
+                TimeSpan.FromSeconds(60),
+                aggregateSeriesRequestOptions);
+
+            await foreach (TimeSeriesPoint point in aggregateSeriesQueryAnalyzer.GetResultsAsync())
+            {
+                long? temperatureCount = (long?)point.GetValue("Count");
+                Console.WriteLine($"{point.Timestamp} - Temperature count: {temperatureCount}");
+            }
+            #endregion Snippet:TimeSeriesInsightsSampleQueryAggregateSeriesWithAggregateVariable
+        }
+
+        private async Task RunQueryEventsSample(TimeSeriesInsightsClient client, TimeSeriesId tsId)
+        {
+            #region Snippet:TimeSeriesInsightsSampleQueryEvents
+            Console.WriteLine("\n\nQuery for raw temperature events over the past 10 minutes.\n");
+
+            // Get events from last 10 minute
+            DateTimeOffset now = DateTime.UtcNow;
+            DateTimeOffset endTime = now.AddMinutes(10);
```

Comment:
If start time is past 10 minutes, shouldn't this value be now?

Reply:
Yes. Let me change that.